File: Frontend/src/types/venta.ts
export interface VentaLinea {
  id: number
  productoId: number
  productoNombre: string
  productoSku: string
  cantidad: number
  unidadMedidaAbreviatura: string
  precioUnitario: number
  descuento: number
  subtotal: number
}

export interface Venta {
  id: number
  sucursalId: number
  sucursalNombre: string
  usuarioId: number
  usuarioNombre: string
  numeroComprobante: string
  clienteNombre: string | null
  clienteEmail: string | null
  clienteTelefono: string | null
  subtotal: number
  descuentoTotal: number
  total: number
  fecha: string
  lineas: VentaLinea[]
}

export interface CrearVentaLinea {
  productoId: number
  unidadMedidaId?: number | null
  cantidad: number
  precioUnitario?: number | null
}

export interface CrearVenta {
  sucursalId: number
  usuarioId: number
  clienteNombre?: string | null
  clienteEmail?: string | null
  clienteTelefono?: string | null
  lineas: CrearVentaLinea[]
}

File: Backend/Models/Venta.cs
namespace InventarioMultiSucursal.Api.Models;

public class Venta
{
    public int Id { get; set; }
    public int SucursalId { get; set; }
    public int UsuarioId { get; set; }
    public string NumeroComprobante { get; set; } = string.Empty;
    public string? ClienteNombre { get; set; }
    public string? ClienteEmail { get; set; }
    public string? ClienteTelefono { get; set; }
    public decimal Subtotal { get; set; }
    public decimal DescuentoTotal { get; set; }
    public decimal Total { get; set; }
    public DateTime Fecha { get; set; }

    public Sucursal? Sucursal { get; set; }
    public Usuario? Usuario { get; set; }
    public ICollection<VentaLinea> Lineas { get; set; } = new List<VentaLinea>();
}
